File: FunctionalHttp.CSharpInterop/Data/MediaTypeExtensions.cs
using FunctionalHttp;

using Microsoft.FSharp.Core;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FunctionalHttp.Interop
{
    public static class MediaTypeExtensions
    {
        public static bool TryGetCharset(this MediaType This, out Charset charset)
        {
            if (FSharpOption<Charset>.get_IsSome(This.Charset))
            {
                charset = This.Charset.Value;
                return true;
            }

            charset = null;
            return false;
        }
    }
}

File: FunctionalHttp/Core.Interop/MediaTypeExtensions.fs
namespace FunctionalHttp.Core.Interop

open FunctionalHttp.Collections
open FunctionalHttp.Core
open System.Runtime.CompilerServices

[<AbstractClass; Sealed; Extension>]
type MediaTypeExtensions private () =
    [<Extension>]
    static member TryGetCharset(this:MediaType, charset : byref<Charset>) = 
        Option.tryGetValue this.Charset &charset

